### PR TITLE
Move dummy backend to Tests namespace

### LINKED ISSUE #14847: Drop class OC_User_HTTP - webdav auth provides the same thing
- [x] https://github.com/owncloud/core/blob/master/lib/private/user/http.php (https://github.com/owncloud/core/pull/14867)

While at it:
- [ ] https://github.com/owncloud/core/blob/master/lib/private/user/dummy.php is for test purpose only -> move to tests/lib
- [x] https://github.com/owncloud/core/blob/master/lib/private/user/example.php is not really needed from my understanding - if at all this shall go into the dev docs (https://github.com/owncloud/core/pull/14880)

### PR DESCRIPTION
**Pull request by LukasReschke**
:warning: :warning: :warning: :warning: :warning: :warning: :warning: :warning:

Fixes https://github.com/owncloud/core/issues/14847

### CODE REVIEW

**Comment by LukasReschke** [comment]
@DeepDiver1975 Please take a look - THX.

File: lib/private/user.php [hunk 1]
```diff
@@ -146,6 +146,10 @@ public static function useBackend($backend = 'database') {
 					self::$_usedBackends[$backend] = new OC_User_Database();
 					self::getManager()->registerBackend(self::$_usedBackends[$backend]);
 					break;
+				case 'dummy':
```
Comment (DeepDiver1975) [inline lib/private/user.php]: bäääh :speak_no_evil:

**Comment by DeepDiver1975** [comment]
fine with me .... an explanation why oracle fails would be nice ....

File: lib/private/user.php (hunk 1 above)
Comment (LukasReschke) [inline lib/private/user.php]: Yup. Thing is otherwise the wrong class name is autoresolved :speak_no_evil:

**Comment by LukasReschke** [comment]
> fine with me .... an explanation why oracle fails would be nice ....

That's a fun story. It's actually failing from time to time on these tests in other branches as well such as master :rocket: :tada: :birthday: 

I guess a retest should help.

**Comment by LukasReschke** [comment]
See for example https://ci.owncloud.org/job/server-stable8.1-linux/database=oci,label=SLAVE/122/

:speak_no_evil: :hear_no_evil: :see_no_evil:

**Comment by LukasReschke** [comment]
Let's track the random fail at https://github.com/owncloud/core/issues/19244

**Comment by LukasReschke** [comment]
And it passed :speak_no_evil: :hear_no_evil: :see_no_evil:

**Comment by DeepDiver1975** [comment]
:+1:

**Comment by PVince81** [comment]
:+1:

**Pull request merged**